drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/ScenarioSimulationEditorPresenter.java

```diff
@@ -360,25 +411,50 @@ protected void open(final String downloadURL) {
         DomGlobal.window.open(downloadURL);
     }
 
-    protected void populateRightPanel() {
-        // Execute only when DatamanagementStrategy already set and  RightPanelPresenter is actually available
+    protected void populateRightDocks(String identifier) {
         if (dataManagementStrategy != null) {
-            getRightPanelPresenter().ifPresent(presenter -> {
-                setRightPanel(presenter);
-            });
+            final PlaceRequest currentRightDockPlaceRequest = getCurrentRightDockPlaceRequest(identifier);
+            switch (identifier) {
+                case SettingsPresenter.IDENTIFIER:
+                    getSettingsPresenter(currentRightDockPlaceRequest).ifPresent(presenter -> {
+                        setSettings(presenter);
+                        presenter.setCurrentPath(path);
```

Comment:
Looking at this with new eyes. 
Now that you have a hold of the dock presenter here, the setup becomes a lot easier if instead of the path you pass in the entire Settings/TestTools/CheatSheet panel. Making these panels fields for the `ScenarioSimulationEditorPresenter`. 
Each dock for this editor then becomes just a placeholder that has a div. This way if you rename the file there is no need for the sync with the Path.

Reply:
@Rikkola 
Somehow I've resolved using AtomicLong (new id for each instance of presenter).
Anyway, the solution you proposed seems much better -I'll use it as soon as required code is in place